spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ResourceBanner.java

```diff
@@ -77,18 +78,25 @@ public void printBanner(Environment environment, Class<?> sourceClass, PrintStre
 	}
 
 	protected List<PropertyResolver> getPropertyResolvers(Environment environment, Class<?> sourceClass) {
-		List<PropertyResolver> resolvers = new ArrayList<>();
-		resolvers.add(environment);
-		resolvers.add(getVersionResolver(sourceClass));
-		resolvers.add(getAnsiResolver());
-		resolvers.add(getTitleResolver(sourceClass));
-		return resolvers;
+		MutablePropertySources propertySources = new MutablePropertySources();
+		propertySources.addLast(getEnvironmentSource(environment));
+		propertySources.addLast(getTitleSource(sourceClass));
+		propertySources.addLast(getAnsiSource());
+		propertySources.addLast(getVersionSource(sourceClass));
+		return Collections.singletonList(new PropertySourcesPropertyResolver(propertySources));
```

Comment:
@wilkinsona Using `singletonList` here changed the behavior of this method by making the returned list unmodifiable. Was this intentional?

For ResourceBanner subclasses, the change to use an unmodifiable list introduced a breaking change. We were able to work around it by creating a new list, adding items from this list, along with our own PropertyResolver(s).

Reply:
Good spot. I've fixed that in a707c5e83e479099f8dd71fb06806e143efdd119